test/codeu/chat/server/DerbyStoreTest.java

```diff
@@ -0,0 +1,190 @@
+package codeu.chat.server;
+
+import static org.junit.Assert.*;
+import org.junit.Test;
+import org.junit.Before;
+
+import codeu.chat.DerbyStore;
+import codeu.chat.common.User;
+import codeu.chat.util.Uuid;
+import codeu.chat.common.BasicController;
+import codeu.chat.common.Conversation;
+import codeu.chat.common.Message;
+
+public class DerbyStoreTest {
+	
+	private DerbyStore ds;
+	private Model model;
+	private BasicController controller;
+		
+	@Before
+	public void doBefore() {
+		ds = new DerbyStore();
+		model = new Model();
+	    controller = new Controller(Uuid.NULL, model);
+	}
+	
+	@Test
+	public void checkUserNameExists() {
+		final User user = controller.newUser("user");
+		boolean userNameExists = false;
+		
+		try {
+			userNameExists = ds.checkUsernameExists(user.name);
+		}
+		catch (Exception ex) {
+			
+		}
+		
+		assertTrue("Check to see if a username already exists",
+				userNameExists);
+	}
+	
+	@Test
+	public void addAndGetUser() {
+		 final User user = controller.newUser("user");
+		 User testUser = null;
+		 try {
+			//ds.addUser(user);
+			 testUser = ds.getUser(user.id);
```

Comment:
I'm using the controller object to create a new conversation, message, or user. The controller has functionality already that saves the object in a Store<Uuid, Object> variable, and I've added code that also puts it into the database. So, every time I make a user object for example it is saved. I'm using DerbyStore to pull that object once its saved.